File: NHAY/NHAY.py
__author__ = "Ashwani Gautam"

import sys


def main():
    flag = 0
    for line in sys.stdin:
        if flag == 0:
            n = int(line.strip())
            flag = 1
            continue
        elif flag == 1:
            needle = line.strip()
            flag = 2
            continue
        else:
            haystack = line.strip()
            flag = 0
        start = -1
        if haystack.find(needle) == -1:
            continue
        while 1:
            if haystack.find(needle, start + 1) == -1:
                break
            else:
                start = haystack.find(needle, start + 1)
                print start
        print
        print

if __name__ == "__main__":
    main()

File: NHAY/NHAY.cpp
#include<iostream>
#include<stdio.h>
#include<string>
using namespace std;
string needle	;
string haystack	;
int main()  {
    long long int t ;
    size_t start	;
    while(scanf ("%lld\n",  &t) == 1){
        	cin >> needle >> haystack	;
        	start = -1	;
        	if (haystack.find(needle,start + 1) == std::string::npos)
        		continue	;
        	while (1) {
        			if (haystack.find(needle,start + 1) == std::string::npos)
        				break	;
        			start = haystack.find(needle,start + 1)	;
        			printf("%zd\n", start)	;
        	}
        	printf("\n\n")	;
        }



    return 0    ;

	}
